include/fastrtps/types/DynamicTypeBuilderPtr.h

```diff
@@ -42,6 +42,16 @@ class DynamicTypeBuilder_ptr : public std::shared_ptr<DynamicTypeBuilder>
     RTPS_DllAPI DynamicTypeBuilder_ptr(DynamicTypeBuilder_ptr&& other) = default;
 
     RTPS_DllAPI DynamicTypeBuilder_ptr& operator=(DynamicTypeBuilder_ptr&&) = default;
+
+    RTPS_DllAPI bool operator!=(std::nullptr_t) const
+    {
+            return bool(*this);
+    }
+
+    RTPS_DllAPI bool operator==(std::nullptr_t) const
+    {
+            return !*this;
+    }
```

Comment:
Adding new API should add corresponding tests.

Reply:
Already checked in \test\unittest\dynamic_types\DynamicTypesTests.cpp